A14-channelz.md

```diff
@@ -0,0 +1,1051 @@
+
+* Number of streams started, succeeded, or failed.
+* Number of gRPC messages sent and received.
+* Number of keepalive messages sent.
+* The last time a outbound stream was started and the last time an inbound
+  stream was started.
+* The last time a outbound message was sent and the last time an inbound
+  message was received.
+* The amount of flow control window granted to the remote endpoint, and the
+  amount of flow control window granted by remote endpoint.
+* All known socket options set.
+
+The stream counts differ slightly than that of the number of calls.  Calls may
+be started before a socket (transport) is available, so the stream counter may
+smaller than the corresponding call counters.  Conversely, if a stream fails and
+the call is retriable, a new stream may be started.  This could make the stream
+counts higher than the call counts.  Streams are considered successful if they
+have the HTTP/2 EoS bit set, or failed if terminated otherwise (usually an
+HTTP/2 RST_STREAM frame).
+
+The message counts represent how many gRPC messages are sent and received.
+Calls may send and receive any number of messages, including zero.
+
+The number of keepalives sent represents how often the local endpoint wants to
+keep the socket alive.  Generally this is an HTTP/2 PING frame.  However, not
+all PING frames are keep alives.  Specifically, pings that were sent as a reply
+(ACK bit set) or were used for reasons unrelated to keep alives should not count
+towards this.  For example, if pings were used to measure network bandwidth,
+they would not be treated as keep alives.   The idea is to measure how often the
+local endpoint is doing extra work to keep the connection alive.
+
+
+Like Channel data, the last time a stream was started by the remote or local
+endpoint gives insight into the activity of the socket.  The number of locally
+started streams is zero for server sockets, as servers do not start streams in
+the gRPC protocol.
+
+In the event of long lived streams, the last time a message was sent or received
+is also useful to know.  The last timestamps are included for messages in both
+directions.
+
+Flow control is a very tricky subject and is often the source of hard to explain
+slowdown.  The amount of window granted by each endpoint to each other is a
+constantly evolving number, which each endpoint sending chunks of window.  For
+the particular flow control windows in the socket data, they are the windows as
+defined by the underlying transport.  For example, HTTP/2 transports report the
+connection level flow control window.  Implementations that don't support flow
+control may set this to the max value, implying that there is no limitation.
+Note that only the connection level flow control window is reported, rather than
+the stream level.  TCP level flow and congestion control counters are exposed
+via socket options.
+
+#### Socket Options
+
+At this low level in the gRPC stack, the implementation difference begin vary
```

Comment:
This sentence is grammatically incorrect and very difficult to parse.  I suggest something like this:

At this low level in the gRPC stack, implementation differences become more significant.